src/rules/no-duplicate-store-ids.ts

```diff
@@ -0,0 +1,47 @@
+import { createEslintRule } from '../utils/rule-creator'
+import { AST_NODE_TYPES } from '@typescript-eslint/utils'
+
+export const RULE_NAME = 'no-duplicate-store-ids'
+export type MESSAGE_IDS = 'duplicatedStoreIds'
+type Options = []
+
+const usingStoreIds = new Set<string>()
+
+export default createEslintRule<Options, MESSAGE_IDS>({
+  name: RULE_NAME,
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Disallow duplicate store ids.'
+    },
+    schema: [],
+    messages: {
+      duplicatedStoreIds: 'No duplicated store ids allowed.'
```

Comment:
Let's add more context here:

```suggestion
      duplicatedStoreIds: 'No duplicated store ids allowed: {{storeId}}'
```